service/src/main/kotlin/fi/espoo/evaka/attendance/StaffAttendanceQueries.kt

```diff
@@ -82,15 +87,27 @@ fun Database.Read.getStaffAttendance(id: StaffAttendanceId): StaffMemberAttendan
 
 fun Database.Read.getExternalStaffAttendance(id: StaffAttendanceExternalId): ExternalStaffMember? = createQuery(
     """
-    SELECT id, name, group_id, arrived
-    FROM staff_attendance_external
+    SELECT
+        sae.id, sae.name, sae.group_id, sae.arrived,
+        EXISTS(
+            SELECT 1 FROM staff_attendance_external osae
+            WHERE osae.name = sae.name
+              AND NOW() < osae.arrived
```

Comment:
Same here, except that you should add a `now` parameter to the function and pass it to SQL.